<a href="https://colab.research.google.com/github/neip313/Python_Machine_Learning/blob/main/Chapter_2_4_eda_data_import.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

---
title: "EDA with Housing Price Prediction - Data Import"
date: 2020-06-07T16:30:47+09:00
tags:
  - "Exploratory Data Analysis"
  - "EDA"
  - "Python"
  - "Kaggle"
categories:
  - "Data Transformation"
  - "Python"
  - "Pandas Merge"
  - "Python Basic"
menu: 
  kaggle:
    name: EDA with Housing Price Prediction - Data Import
---

![](/img/kaggle/housing_price/eda_1/tutorial_01.png)

## I. 개요

- 이제 본격적으로 Kaggle 데이터를 활용하여 분석을 진행한다. 
- 데이터는 이미 다운 받은 상태를 전제로 하며, 만약에 데이터가 없다면 이전 포스팅에서 절차를 확인하기 바란다. (미리보기 가능)
  + [캐글 데이터 다운로드 받기 (via Colab)](https://chloevan.github.io/kaggle/colab_with_drive/)

## II. 구글 드라이브 연동
- 구글 코랩을 시작하면 언제든지 가장 먼저 해야 하는 것은 드라이브 연동이다. 

In [ ]:
from google.colab import drive # 패키지 불러오기 
from os.path import join  

ROOT = "/content/drive"     # 드라이브 기본 경로
print(ROOT)                 # print content of ROOT (Optional)
drive.mount(ROOT)           # 드라이브 기본 경로 Mount

MY_GOOGLE_DRIVE_PATH = 'My Drive/Colab Notebooks/inflearn_kaggle/' # 프로젝트 경로
PROJECT_PATH = join(ROOT, MY_GOOGLE_DRIVE_PATH) # 프로젝트 경로
print(PROJECT_PATH)

/content/drive
Mounted at /content/drive
/content/drive/My Drive/Colab Notebooks/inflearn_kaggle/


In [ ]:
%cd "{PROJECT_PATH}"

/content/drive/My Drive/Colab Notebooks/inflearn_kaggle


- 위 코드가 에러 없이 돌아간다면 이제 데이터를 불러올 차례다. 


In [ ]:
!ls

data  docs  source


- 필자는 `inflearn_kaggle` 폴더안에 `data`, `docs`, `source` 등의 하위 폴더를 추가로 만들었다. 
- 즉, `data` 안에 다운로드 받은 파일이 있을 것이다. 

## III. 캐글 데이터 수집 및 EDA 
- 우선 데이터를 수집하기에 앞서서 `EDA`에 관한 필수 패키지를 설치하자. 

In [ ]:
import pandas as pd  # 데이터 가공, 변환(dplyr)
import pandas_profiling # 보고서 기능 # 아나콘다 할 때... 실습
import numpy as np # 수치 연산 & 배열, 행렬
import matplotlib as mpl # 시각화
import matplotlib.pyplot as plt # 시각화 
import seaborn as sns # 시각화

from IPython.core.display import display, HTML

### (1) 데이터 수집
- 지난 시간에 받은 데이터가 총 4개임을 확인했다.
  + data_description.txt
  + sample_submission.csv
  + test.csv
  + train.csv
- 여기에서는 우선 `test.csv` & `train.csv` 파일을 받도록 한다. 

In [ ]:
train = pd.read_csv('data/train.csv')
test = pd.read_csv('data/test.csv')
print("data import is done")

data import is done


### (2) 데이터 확인
- `Kaggle` 데이터를 불러오면 우선 확인해야 하는 것은 데이터셋의 크기다. 
  + 변수의 갯수
  + Numeric 변수 & Categorical 변수의 개수 등을 파악해야 한다.
- Point 1 - `train`데이터에서 굳이 훈련데이터와 테스트 데이터를 구분할 필요는 없다. 
  + 보통 `Kaggle`에서는 테스트 데이터를 주기적으로 업데이트 해준다.
- Point 2 - 보통 `test` 데이터의 변수의 개수가 하나 더 작다. 


In [ ]:
train.shape, test.shape

((1460, 81), (1459, 80))

- 그 후 `train`데이터의 `상위 5개`의 데이터만 확인한다. 

In [ ]:
display(train.head())

Id  MSSubClass MSZoning  ...  SaleType  SaleCondition SalePrice
0   1          60       RL  ...        WD         Normal    208500
1   2          20       RL  ...        WD         Normal    181500
2   3          60       RL  ...        WD         Normal    223500
3   4          70       RL  ...        WD        Abnorml    140000
4   5          60       RL  ...        WD         Normal    250000

[5 rows x 81 columns]

- 그 다음 확인해야 하는 것은 `Numerical` 변수와 `Categorical` 변수를 구분한다. 
  + 먼저 `numerical_features`를 구분하자.

In [ ]:
numeric_features = train.select_dtypes(include=[np.number]) # 수치형 데이터
print(numeric_features.columns)
print("The total number of numeric features are: ", len(numeric_features.columns))

Index(['Id', 'MSSubClass', 'LotFrontage', 'LotArea', 'OverallQual',
       'OverallCond', 'YearBuilt', 'YearRemodAdd', 'MasVnrArea', 'BsmtFinSF1',
       'BsmtFinSF2', 'BsmtUnfSF', 'TotalBsmtSF', '1stFlrSF', '2ndFlrSF',
       'LowQualFinSF', 'GrLivArea', 'BsmtFullBath', 'BsmtHalfBath', 'FullBath',
       'HalfBath', 'BedroomAbvGr', 'KitchenAbvGr', 'TotRmsAbvGrd',
       'Fireplaces', 'GarageYrBlt', 'GarageCars', 'GarageArea', 'WoodDeckSF',
       'OpenPorchSF', 'EnclosedPorch', '3SsnPorch', 'ScreenPorch', 'PoolArea',
       'MiscVal', 'MoSold', 'YrSold', 'SalePrice'],
      dtype='object')
The total number of numeric features are:  38


- `numeric_features`을 제외한 나머지 변수를 추출하자.

In [ ]:
categorical_features = train.select_dtypes(exclude=[np.number])
print(categorical_features.columns)
print("The total number of numeric features are: ", len(categorical_features.columns))

Index(['MSZoning', 'Street', 'Alley', 'LotShape', 'LandContour', 'Utilities',
       'LotConfig', 'LandSlope', 'Neighborhood', 'Condition1', 'Condition2',
       'BldgType', 'HouseStyle', 'RoofStyle', 'RoofMatl', 'Exterior1st',
       'Exterior2nd', 'MasVnrType', 'ExterQual', 'ExterCond', 'Foundation',
       'BsmtQual', 'BsmtCond', 'BsmtExposure', 'BsmtFinType1', 'BsmtFinType2',
       'Heating', 'HeatingQC', 'CentralAir', 'Electrical', 'KitchenQual',
       'Functional', 'FireplaceQu', 'GarageType', 'GarageFinish', 'GarageQual',
       'GarageCond', 'PavedDrive', 'PoolQC', 'Fence', 'MiscFeature',
       'SaleType', 'SaleCondition'],
      dtype='object')
The total number of numeric features are:  43


## IV. Review & What's next
- 우선 전체 데이터는 `1460`개 변수는 `81`개로 확인했다. 
  + 그 중 수치형 변수는 38개, 문자형 변수는 43개인것으로 확인된다. 
  + 그런데, 이 수치를 믿어야 할까?
- Stay Tuned